### ISSUE TITLE
RAPTOR-API

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#2684]
missing: API support for expressing masters and workers

relate to DDMD evolution

**Comment by mturilli** [comment]
Discussion seems to have converged. Need to start to implement this. Please add checkbox list to this ticket so that we can trace the implementation process.

**Comment by mtitov** [comment]
This will be part of the `TaskDescription` refactoring effort

- [ ] Create base class for `TaskDescription` and/or `Task`
   - [ ] Create a class for `Master` tasks as `RaptorMaster`
   - [ ] Create a separate class for `Worker` tasks, or represent them as regular tasks of the `function` type and with a reference to a corresponding master-task in the `scheduler` attribute (?)
- [ ] Create a `Pilot` method to submit master-tasks, e.g., `pilot.submit_raptor(...)`

p.s. ref. to the original discussion - [link](https://github.com/radical-cybertools/radical.pilot/discussions/2717)

**Comment by andre-merzky** [comment]
> Create a separate class for Worker tasks, or [...]

I would suggest the first one, for consistency with other task classes.  The `scheduler` entry would be nice to express differently on the API, like this maybe:
```
pmgr = ...
pilot = rp.Pilot(...)
master = rp.raptor.Master(...)
worker = rp.raptor.Worker(...)
todo = rp.FunctionTask(...)

pmgr.launch(pilot)  # or keep `pmgr.submit_pilots()`
pilot.submit(master)
master.submit(worker)
master.submit(todo)
```

> Create a Pilot method to submit master-tasks, e.g., pilot.submit_raptor(...)

I would suggest to expand this to submit *any* task (not only master tasks) to the pilot.  In terms of implementation this is trivial, and it makes the API more consistent.  Semantically it is the same as specifying the `pilot` attribute in the task description (early binding).

**Comment by mtitov** [comment]
@andre-merzky yeah, agree with your proposal, and I tried to sum up issues/actions to follow (as general descriptions), thus please feel free to edit that list accordingly

**Comment by andre-merzky** [comment]
> @andre-merzky yeah, agree with your proposal, and I tried to sum up issues/actions to follow (as general descriptions), thus please feel free to edit that list accordingly

Ack, and thanks for jump-starting this.  I'll edit the original comment then.

**Comment by andre-merzky** [comment]
This is addressed in #2892
